rita/src/rita_common/network_endpoints/mod.rs

```diff
@@ -53,25 +53,20 @@ pub fn hello_response(
     their_id: Json<serde_json::Value>,
     req: HttpRequest,
 ) -> Box<Future<Item = Json<LocalIdentity>, Error = Error>> {
-    let new_id: Result<Identity, serde_json::Error> = serde_json::from_value(their_id.clone());
-    let old_id: Result<LocalIdentity, serde_json::Error> = serde_json::from_value(their_id.clone());
+    let id: Result<Identity, serde_json::Error> = serde_json::from_value(their_id.clone());
 
-    match new_id {
+    match id {
         Ok(new_id) => {
             info!("sending to new hello handler");
-            hello_response_new(new_id, req)
+            hello_response_handler(new_id, req)
```

Comment:
maybe we should inline this?

Reply:
should we just remove hello_response in general and do the deserialization in the calling function?